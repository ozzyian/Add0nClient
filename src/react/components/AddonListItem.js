import React from 'react';
const ipc = require('electron').ipcRenderer;
/**
 *
 */
class AddonListItem extends React.Component {
  /**
   *
   * @param {*} props
   */
  constructor(props) {
    super(props);
    this.updateAddon = this.updateAddon.bind(this);
    this.addon = this.props.addon;
    this.state = {
      updated: false,
      upateAvailable: true,
    };
  }

  /**
   *
   */
  componentDidMount() {
    ipc.on('update' + this.addon.id, () => {
      this.setState({updated: true});
    });
  }

  /**
   *
   */
  updateAddon() {
    ipc.send('update', this.addon);
  }
  /**
   * @return {*}
   */
  render() {
    if (this.state.updated) {
      return (
        <tr>
          <td>{this.props.addon.id}</td>
          <td>{this.props.addon.name}</td>
          <td>Updated</td>
          <td>{this.props.addon.gameVersionLatestFiles[0].projectFileName}</td>
          <td>{this.props.addon.authors[0].name}</td>
        </tr>
      );
    } else {
      return (
        <tr>
          <td>{this.props.addon.id}</td>
          <td>{this.props.addon.name}</td>
          <td>
            <button onClick={this.updateAddon}>Update</button>
          </td>
          <td>{this.props.addon.gameVersionLatestFiles[0].projectFileName}</td>
          <td>{this.props.addon.authors[0].name}</td>
        </tr>
      );
    }
  }
}

export default AddonListItem;
