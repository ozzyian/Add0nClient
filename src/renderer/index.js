// Initial welcome page. Delete the following line to remove it.
const body = document.getElementsByTagName('body')[0];

if (window.matchMedia('(prefers-color-scheme:dark)').matches) {
  body.style.color = 'white';
}

const header = document.createElement('h1');
header.innerText = 'En header';
body.appendChild(header);
