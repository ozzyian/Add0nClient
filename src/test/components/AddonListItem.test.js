import React from 'react';
import AddonListItem from '../../react/components/AddonListItem';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
describe('Testing the component', () => {
  it('Renders the snapshot', () => {
    const tree = renderer.create(<AddonListItem addons={['1']} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
