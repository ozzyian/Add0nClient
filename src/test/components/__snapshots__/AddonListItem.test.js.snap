// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the snapshot 1`] = `
<div
  className="list-group-item"
  disabled={false}
  onClick={[Function]}
>
  id: 
  
</div>
`;
